docs/local_executor_run.md

```diff
@@ -0,0 +1,154 @@
+# Local Executor state diagram
+
+This state diagram describes the amalgamation of `local_executor.py` and `run.py` to facilitate understanding changes that cut across both of them. Details should be verified with the current version of the code, and also cross-referenced with StubExecutor & the pytests.
+
+```mermaid
```

Comment:
for some reason this doesn't render on github, a view of it is here: https://mermaid.live/edit#pako:eNrdWG1v2jAQ_iuWP8FEWRJeE1WtGISJjaUVjHXrmKI0MZA2xMhOqtKq_33OCySGQEnX8mH5hO-eO989sS93PEETWwgqkHqGhzq2MSXGfOwC9oQSUQGXyLVsd5oSSkxI0MIgyEpJKwpQH5Dpezy4uhJz4JoCurZrOPYjJ64r4Au-AUPfNBGyOFWD-SEEk5SoqYC24ZrIcZAFCquQiimEzCHW4aUhohDt2TVsh9tQFHn3EQ2cKSPiyrADj6DwwyA29inTRwgXewgQezrzAJ7EvEWa4LnHjj9HAD3Y1KOJ2MSuZ9guIsDCiAY-IkiCuMU35dAZGIw0rad93lb5VA9eKijQGfYdC9ygIrgcqJetgdpJ0BGyHHGCyTD0uQ2bIk-PoIXii2BzTdfEcChaRhpGXMgGx4yDJgkxlRzEbGpyEKL-VNuj7xwkk4UM3B4aMtCv5aF6TB5ePAzax1YOAt7iHNSOk3-3p7X6vesXCcjA7aFhA52ZLV8P6lvp7rr1eerCcNRuq2onHfUmARmQTAJG2lft4ko7Pyj_bXCOY8AT0zjOOTh_qwswGFwMcqadkXXzuJ-Hdktrq_2-2tE__dJHQ3VwUFV83WXgafGI780OPw3ycU7Du_Kxr0bmpEMU3qVsdFu9_r6akZeeuBzkKR1vQY74z-TQ_4Sd3x_-gJOTs3Ufz0IGJkFswVrCZbA8oYjco3RHLaYsJK7TTSlWG2xasG54EfXghSKHSRmzIQFF00ChmI1gTb2HyJwNB17sJ4WLjW3s6hPWr-sTTO70SHN6as6wbaKzs_RMEvrdaRQTt9PnOqhqTB-eLxzksdNCgwmF0onvOMuDvTQiLygYY-hWlKuBZSP7JHdGrj6JhyaW7i223SDZ9KwVOkrj0gPUPmUzCaGWnp922HChJJYKWAm3YsfU000HGa6_CLnJfl-1aL9NsALi1erQbLtbB8FGyOR-2JT7Bu8xC8ZAzm51tXbcDnH3i6qHMHb9uCkzSyiuhLAE58ybYVtsFH8KIGPozdAcjaHCfloGuRvDsfvMcAYrDMOla0IlzKwE_YWVjO5r6cJwrzFma5ZHtITKE3yAyolQrlYrjXqTPQ1RlCW5UoJLqIhlqdKU5UZdbjQrgiQI0nMJPoYuxLJQr1WkuixJYlMShcACWTYrUN-iPw_C_xCe_wKXTCtb